### PR TITLE
[devtools] Property keys should expand it's children

### PR DESCRIPTION
**Pull request by TryingToImprove**
## Summary

_(this is the same as #20427 but with the correct implementation)_

When inspecting components in React Devtools the clickarea of toggling a complex object is very small (only the arrow). This PR will make the property-name clickable.

It is using the same approach as with https://github.com/facebook/react/blob/master/packages/react-devtools-shared/src/devtools/views/Components/InspectedElementHooksTree.js#L228-L236

The yellow parts will be clickable with this PR.

image

## Test Plan

1. Open dev-tools in this branch
2. Inspect a element
3. Click a property name and see that it is toggleable

### CODE REVIEW

File: packages/react-devtools-shared/src/devtools/views/Components/KeyValue.js [hunk 1]
```diff
@@ -398,8 +408,12 @@ export default function KeyValue({
           ) : (
             <div className={styles.ExpandCollapseToggleSpacer} />
           )}
-          {renderedName}
-          <div className={styles.AfterName}>:</div>
+          <span
```
Comment (rheh) [inline packages/react-devtools-shared/src/devtools/views/Components/KeyValue.js:411]: This span and its inner jsx is repeated three/four times.  Should this be refactored into the RenderedName component? Or a new RenderedNameContainer?

**Review by rheh — COMMENTED** [review]
One suggestion to remove repetition.

**Comment by bvaughn** [comment]
This looks like a nice UX improvement in general, but there's a case it regresses on (I think): the editable key case. IMO clicking on a key to edit it should not expand/collapse its value. Here's a repro:

Regression

**Comment by bvaughn** [comment]
> When inspecting components in React Devtools the clickarea of toggling a complex object is very small (only the arrow). This PR will make the property-name clickable.

Just to be clear too, clicking the value string to the right also toggles open.

**Comment by TryingToImprove** [comment]
@bvaughn  never noticed it was possible to edit the keys. :)

Would something like double-clicking the key to make it editable be better? 

Looking at the sources tab in Chrome it is not possible to edit a variable name there when debugging. The same goes for the debug panel in VS Code.  Both of them supports collapse/extend when clicking the keys. 

In Chrome you double-click on the value to edit it; In VS Code you can double click on either key or value to edit the value.

**Comment by bvaughn** [comment]
> Would something like double-clicking the key to make it editable be better?

No. I don't think double-clicking is a very intuitive gesture.

I've designed the props/state/hooks editable interface to work like Chrome's style editor in the Elements panel. I think this is a pretty familiar interface for most web developers.

> Looking at the sources tab in Chrome it is not possible to edit a variable name there when debugging. The same goes for the debug panel in VS Code. Both of them supports collapse/extend when clicking the keys.

I don't think it makes sense conceptually to edit a variable name in the sources panel. (What would that even do?)

> In Chrome you double-click on the value to edit it; In VS Code you can double click on either key or value to edit the value.

I've designed the props/state/hooks editable interface to work like Chrome's style editor in the Elements panel. I think this is a pretty familiar interface for most web developers. That panel allows you to start editing by focusing on either a key or value field.

**Comment by TryingToImprove** [comment]
> I don't think it makes sense conceptually to edit a variable name in the sources panel. (What would that even do?)

I think it would do the same as the dev-tools do now - allowing to change a property on a object "on the fly".

> I've designed the props/state/hooks editable interface to work like Chrome's style editor in the Elements panel. I think this is a pretty familiar interface for most web developers. That panel allows you to start editing by focusing on either a key or value field.

I think that it is a valid point, however the Elements panel does not support a tree view like the Debug and dev-tools does. 

Further I will argue that it seems a little off that the user is able to click on a value of a object and see it expanded vs. not enable editing it like the primitive values (strings, numbers, etc.).

**Comment by TryingToImprove** [comment]
I agree with the regression and another solution should be made. I will close this PR and create a issue instead. :)

**Comment by bvaughn** [comment]
Thanks @TryingToImprove

**Pull request closed without merging**